src/elo/pipeline/index.ts

```diff
@@ -0,0 +1,61 @@
+import { GraphQLContext } from "../../context";
+import { pullComputedElosInRange, pullMatches, writeComputedElos, writeEloSnapshots } from "../util/dbInterface";
+import { calculate } from "./calculate";
+import { getComputedEloFromMatches } from "../util/extract";
+import { updateComputedElo } from "./update";
+
+
+export const eloPipeline = async (
+  oldestMatchId: string,
+  context: GraphQLContext
+) => {
+
+  // pull matches from database
+  const matches = await pullMatches(oldestMatchId, context);
+
+  // pull computed elos from database based on matches
+  const computedElos = await getComputedEloFromMatches(matches, context);
+
+  // run the calculation routine!
+  const [eloSnapshots, create] = calculate(matches, computedElos, context);
+
+  // min is 0 if we're creating new entires, because we have to update
+  // ranks of everything _after_ the new bois
+  const min = create ? 0 : (computedElos.get("min")?.elo || 0);
```

Comment:
Super minor, but I think this could be:

```
const min = !create ? computedElos.get("min")?.elo || 0;
```

Reply:
That is clearer, you're right. Thanks for the suggestion!

Reply:
Oop, so I actually misread the code, and I'm not sure yours will work - the `|| 0` at the end of the last part of the tertiary operator is enforced by the type system, just in case `computedElos.get("min")` is undefined - it definitionally won't be, but that's not known to the transpiler, so the `|| 0` is in there to appease it, and is separate from the tertiary operation itself. The best I could do would be to switch the order:

```ts
const min = !create ? (computedElos.get("min")?.elo || 0) : 0
```